org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/AbstractXtextIntegrationTest.xtend

```diff
@@ -66,18 +35,59 @@ abstract class AbstractXtextIntegrationTest {
 		return builder.sign(Algorithm.HMAC256("secret"))
 	}
 
+	/**
+	 * Workaround related to the following dropwizard / dropwizard-guice / jersey2-guice issues
+	 * https://github.com/dropwizard/dropwizard/issues/1772
+	 * https://github.com/HubSpot/dropwizard-guice/issues/95
+	 * https://github.com/HubSpot/dropwizard-guice/issues/88
+	 * https://github.com/Squarespace/jersey2-guice/pull/39
+	 */
+	static def void ensureServiceLocatorPopulated() {
+		JerseyGuiceUtils.reset
+	}
+
+	static var DummyGlobalScopeResource dummyResource
+
+	/**
+	 * Actually a client rule, but acts as the server (remote) for this test
+	 */
 	@Rule
-	public val dropwizardAppRule = new DropwizardAppRule(
-		TestEditorTestApplication,
-		ResourceHelpers.resourceFilePath('test-config.yml'),
-		configs
-	)
+	public var DropwizardClientRule dropwizardServerRule
+
+	@Rule
+	public var DropwizardAppRule<TestEditorConfiguration> dropwizardClientRule = createClientRule
+
+	private def createServerRule() {
+		dummyResource = new DummyGlobalScopeResource
+		ensureServiceLocatorPopulated
+		dropwizardServerRule = new EagerDropwizardClientRule(dummyResource)
+
+		return dropwizardServerRule
+	}
+
+	private def createClientRule() {
+		dropwizardClientRule = new DropwizardAppRule(TestEditorApplication,
+			ResourceHelpers.resourceFilePath('test-config.yml'),
+			#[config('indexServiceURL', '''«createServerRule.baseUri»/xtext/index/global-scope''')])
```

Comment:
That the server rule is created here and is setting dropwizardServerRule is a bit hidden here.
How a method 'initializeRules' that creates the rules and initializes the variables.